shared/js/media/media_frame.js

```diff
@@ -157,6 +176,11 @@ MediaFrame.prototype.displayImage = function displayImage(blob,
   // Keep track of what kind of content we have
   this.displayingImage = true;
 
+  // If a locale is present and ready, go ahead and localize now.
+  if (navigator.mozL10n.readyState === 'complete') {
```

Comment:
You might add...

// Otherwise, localization will be handled by the ready() callback above

or something like that.